enhancements/fbc-run-bundle.md

```diff
@@ -0,0 +1,268 @@
+
+2. The Operator SDK, using the data gathered from the above step as parameters, will invoke Operator Registry APIs that will generate three major components in the form of Operator Registry-defined objects - Package, Channel & Bundle.
+
+3. The Operator SDK will convert these objects into more commonly-consumable formats like JSON/YAML blobs and combine these blobs together to generate a basic file-based catalog representation of the bundle image. In other words, the simplest file-based catalog consists of one package blob, one channel blob, and one bundle blob. These blobs represent the bundle data.
+
+4. If an index image was passed in as a parameter through the `--index-image` flag of the run bundle command, a series of additional steps take place:
+
+    - The index image will be rendered into a declarative config object, similar to the bundle image.
+    - Operator SDK will check if the bundle image is already present in the index image.
+        - If the bundle image is already present in the index image, Operator SDK will not add the bundle to the index but rather error out.
+        - If the bundle is not present in the index image, Operator SDK will add the contents of the bundle image to the existing index image and serve the resulting index image (with the bundle added to it).
+
+    - Note: If the index image that was passed is in SQLite format, Operator SDK will use a different container creation command that uses `opm registry add` to add the bundle to the index and `opm registry serve` to serve the index which is a database file on created on the pod.
+
+5. The Operator SDK will invoke more Operator Registry APIs to ensure that the FBC that was generated from the above steps is indeed a valid FBC. This steps essentially validates the generated FBC by converting the declarative config object to a model object which internally calls the API-equivalent of the `opm validate <FBC>` command.
+
+6. Three unique data points will be stored in the `IndexImageCatalogCreator` object that will eventually be used in the ephemeral pod:
+
+    - FBC directory name where the FBC file resides in → `FBCDir`
+    - FBC file name that has all the contents to be served through the pod → `FBCFile`
+    - Contents of file-based catalog file (string YAML/JSON) → `FBCContent`
+
+7. During the creation of the ephemeral pod (which is used to serve the file-based catalog to gRPC), the Operator SDK will invoke a set of CLI commands to create a directory and file on the pod during runtime.
+
+8. The contents of the generated File-Based Catalog will be written to the file that is created on the ephemeral pod using the following command template -
+
+    ```go
+    `mkdir -p {{ .FBCDir }} && \
+    echo '{{ .FBCContent }}' >> {{ .FBCFile }} && \
+    opm serve {{ .FBCDir }} -p {{ .GRPCPort }}`
```

Comment:
Why are we using OPM serve?

Should we not do here the same steps described in the doc: https://docs.openshift.com/container-platform/4.9/operators/admin/olm-managing-custom-catalogs.html#olm-creating-fb-catalog-image_olm-managing-custom-catalogs

Reply:
@camilamacedo86 in [`install.go`](https://github.com/rashmigottipati/operator-sdk/blob/run-bundle-prototype/internal/olm/operator/bundle/install.go), we do create the FBC from scratch and add the bundle. Eventually, we have to serve the FBC to gRPC for the installation. Since commands like `opm registry serve` are going away, we are using the new command `opm serve`.

Reply:
that makes sense.